connaisseur/config.py

```diff
@@ -0,0 +1,155 @@
+import os
+import json
+import yaml
+from jsonschema import validate, ValidationError
+from connaisseur.exceptions import NotFoundException, InvalidFormatException
+
+
+class Config:
+    """
+    Config Object, that contains all notary configuration inside a list.
+    """
+
+    CONFIG_PATH = "/etc/connaisseur-config/config.yaml"
+    CONFIG_SCHEMA_PATH = "/app/connaisseur/res/config_schema.json"
+    notaries: list = []
+
+    def __init__(self):
+        """
+        Creates a Config object, containing all notary configurations. It does so by
+        reading a config file, doing input validation and then creating Notary object,
+        storing them in a list.
+
+        Raises `NotFoundException` if the configuration file is not found.
+
+        Raises `InvalidFormatException` if the configuration file has an invalid format.
+        """
+        with open(self.CONFIG_PATH, "r") as configfile:
+            config_content = yaml.safe_load(configfile)
+
+        if not config_content:
+            raise NotFoundException("error getting any notary host configurations.")
+
+        with open(self.CONFIG_SCHEMA_PATH, "r") as schema_file:
+            schema = json.load(schema_file)
+
+        try:
+            validate(instance=config_content, schema=schema)
+        except ValidationError:
+            raise InvalidFormatException(
+                "invalid format for connaisseur configuration."
+            )
+
+        self.notaries = [Notary(notary) for notary in config_content]
+
+    def get_notary(self, notary_name: str = None):
+        """
+        Returns the notary configuration with the given `notary_name`. If `notary_name`
+        is None, the top most element of the notary configuration list is returned.
+
+        Raises `NotFoundException` if no top most element can be found.
+        """
+        try:
+            if notary_name:
+                return next(
+                    notary for notary in self.notaries if notary.name == notary_name
+                )
+            return next(iter(self.notaries))
+        except StopIteration:
+            raise NotFoundException(
+                "the given notary configuration could not be found.",
+                {"notary_name": notary_name},
+            )
+
+
+class Notary:  # pylint: disable=too-many-instance-attributes
```

Comment:
Would this class be better suited for a separate file?

Reply:
Yea unsure about that, i found the `Notary` class to be logically close enough to the `Config` class, that I put both in one file. If you + maybe peter don't think this is rather important, I would leave it as is.

Reply:
I have no stakes in this.